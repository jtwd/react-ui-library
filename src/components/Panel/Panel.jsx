import React from 'react'
import { node, bool } from 'prop-types'

import { PanelWrapper, Body, Header, Footer } from './Panel.styles'

function Panel ({header, footer, footerRight, children, primary, secondary, danger, success, ...props}) {
  return (
    <PanelWrapper danger={danger} success={success} primary={primary} secondary={secondary} collapse {...props}>
      {header && (
        <Header danger={danger} success={success} primary={primary} secondary={secondary}>{header}</Header>
      )}
      <Body collapse>
        {children}
      </Body>
      {footer && (
        <Footer danger={danger} success={success} primary={primary} secondary={secondary} right={footerRight}>{footer}</Footer>
      )}
    </PanelWrapper>
  )
}

Panel.propTypes = {
  primary: bool,
  secondary: bool,
  danger: bool,
  success: bool,
  /** Header content */
  header: node,
  /** Footer content */
  footer: node,
  /** Footer right align */
  footerRight: bool,
  /** Panel content */
  children: node
}

Panel.defaultProps = {
  primary: false,
  secondary: false,
  danger: false,
  success: false,
  header: null,
  footer: null,
  footerRight: false,
  children: null
}

export default Panel
