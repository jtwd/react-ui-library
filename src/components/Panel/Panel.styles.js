import styled from 'styled-components'
import { md, sm, xs, xxs } from '../../theme/spacers'
import { transWhiteLight, transBlackLight } from '../../theme/colors'
import { radiusRound, borderLine, primaryBold } from '../../theme/units'
import boxShadowHover from '../../theme/mixins/boxShadowHover'

const headFootStyles = `
  padding: ${xs} ${xs} ${xs} ${md};
  background: ${transBlackLight};
  
  > * {
    &:first-child {
      margin-top: 0;
    }
    &:last-child {
      margin-bottom: 0;
    }
  }
`

export const PanelWrapper = styled.div`
  margin-bottom: ${md};
  border-radius: ${radiusRound};
  background: ${transWhiteLight};
  position: relative;
  border: ${borderLine};
  ${boxShadowHover}
`

export const Body = styled.div`
  padding: ${sm} ${md};
`

export const Header = styled.div`
  ${headFootStyles};
  border-bottom: ${borderLine};
  font-weight: ${primaryBold};
  margin-bottom: ${xxs};
  border-radius: ${radiusRound} ${radiusRound} 0 0;
`

export const Footer = styled.div`
  ${headFootStyles};
  border-top: ${borderLine};
  margin-top: ${xxs};
  border-radius: 0 0 ${radiusRound} ${radiusRound};
  
  ${props => props.right && `
    text-align: right;
  `}
`