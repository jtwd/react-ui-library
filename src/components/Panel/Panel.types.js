import { trans, alert, confirm, primary, secondary } from '../_theme/palette'
import { borderLine } from "../_theme/units"

export const panel = {
  default: `
    border: ${borderLine.default};
    background: ${trans(1, false)};
  `,
  primary: `
    border: ${borderLine.primary};
    background: ${trans(1, false)};
    color: ${primary('light')};
  `,
  secondary: `
    border: ${borderLine.secondary};
    background: ${trans(1, false)};
    color: ${secondary('light')};
  `,
  danger: `
    border: ${borderLine.danger};
    background: ${alert('dark')};
  `,
  success: `
    border: ${borderLine.success};
    background: ${confirm('dark')};
  `,
}
export const header = {
  default: `
    border-bottom: ${borderLine.default};
    background: ${trans(1)};
  `,
  primary: `
    background: ${trans(1)};
    color: ${primary()};
  `,
  secondary: `
    background: ${trans(1)};
    color: ${secondary('light')};
  `,
  danger: `
    background: ${trans(1)};
  `,
  success: `
    background: ${trans(1)};
  `,
}
export const footer = {
  default: `
    background: ${trans(1)};
    border-top: ${borderLine.default};
  `,
  primary: `
    background: ${trans(1)};
  `,
  secondary: `
    background: ${trans(1)};
  `,
  danger: `
    background: ${trans(1)};
  `,
  success: `
    background: ${trans(1)};
  `
}
