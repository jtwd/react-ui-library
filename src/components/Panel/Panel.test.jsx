import React from 'react'
import renderer from 'react-test-renderer'
import Panel from './Panel'

test('Panel renders correctly by default', () => {
  const component = renderer.create(<Panel>Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel renders header', () => {
  const component = renderer.create(<Panel header="Panel header">Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel renders header when passed a component', () => {
  const Title = <span>Panel header <strong>component</strong></span>
  const component = renderer.create(<Panel header={Title}>Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel renders footer', () => {
  const component = renderer.create(<Panel footer="Panel footer">Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel renders footer when passed a component', () => {
  const Title = <span>Panel footer <strong>component</strong></span>
  const component = renderer.create(<Panel footer={Title}>Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel aligns footer content to the right', () => {
  const component = renderer.create(<Panel footer="Panel footer" footerRight>Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Panel renders a header and footer', () => {
  const component = renderer.create(<Panel header="Panel header" footer="Panel footer">Panel content</Panel>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
