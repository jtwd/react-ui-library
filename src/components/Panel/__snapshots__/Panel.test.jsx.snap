// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel aligns footer content to the right 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
  <div
    className="sc-ifAKCX eVdxlc"
  >
    Panel footer
  </div>
</div>
`;

exports[`Panel renders a header and footer 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="sc-bxivhb dZLVUl"
  >
    Panel header
  </div>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
  <div
    className="sc-ifAKCX AMMvi"
  >
    Panel footer
  </div>
</div>
`;

exports[`Panel renders correctly by default 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
</div>
`;

exports[`Panel renders footer 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
  <div
    className="sc-ifAKCX AMMvi"
  >
    Panel footer
  </div>
</div>
`;

exports[`Panel renders footer when passed a component 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
  <div
    className="sc-ifAKCX AMMvi"
  >
    <span>
      Panel footer 
      <strong>
        component
      </strong>
    </span>
  </div>
</div>
`;

exports[`Panel renders header 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="sc-bxivhb dZLVUl"
  >
    Panel header
  </div>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
</div>
`;

exports[`Panel renders header when passed a component 1`] = `
<div
  className="sc-bwzfXH bVxRpy sc-bdVaJa igwadd"
>
  <div
    className="sc-bxivhb dZLVUl"
  >
    <span>
      Panel header 
      <strong>
        component
      </strong>
    </span>
  </div>
  <div
    className="Panel-body sc-htpNat hKzHqg"
  >
    Panel content
  </div>
</div>
`;
